calculator_test.go

```diff
@@ -2,23 +2,200 @@ package calculator_test
+		{nums: []float64{1, 8, 10}, want: -17, name: "1 - 8 - 10 = -17"},
+	}
+
+	for _, tc := range testCases {
+		got := calculator.Subtract(tc.nums...)
+
+		if tc.want != got {
+			t.Errorf("Error: unexpected value returned - want %.1f, got %.1f", tc.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+
+	testCases := []variadicTestCase{
+		{nums: []float64{3, 2, 1}, want: 6, name: "3 * 2 * 1 = 6"},
+		{nums: []float64{100, 8, 12}, want: 9600, name: "100 * 8 * 12 = 9600"},
+		{nums: []float64{-2, 8, 10}, want: -160, name: "-1 * 8 * 10 = -160"},
+		{nums: []float64{400, 13, 0, 4}, want: 0, name: "400 * 13 * 0 * 4 = 0"},
+	}
+
+	for _, tc := range testCases {
+		got := calculator.Multiply(tc.nums...)
+
+		if tc.want != got {
+			t.Errorf("Error: unexpected value returned - want %.1f, got %.1f", tc.want, got)
+		}
+	}
+
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+
+	testCases := []variadicTestCase{
+		{nums: []float64{7, 0, 10, 44}, want: 999, name: "1 / 8 / -10 = -0.0125", errExpected: true},
+		{nums: []float64{3, 2}, want: 1.5, name: "3 / 2 = 1.5", errExpected: false},
+		{nums: []float64{100, 8, 12}, want: 1.0416666666666667, name: "100 / 8 / 12 = 1.041666667", errExpected: false},
+		{nums: []float64{1, 8, -10}, want: -0.0125, name: "1 / 8 / -10 = -0.0125", errExpected: false},
+		{nums: []float64{10}, want: 10, name: "10 - single value passed", errExpected: false},
+	}
+
+	for _, tc := range testCases {
+		got, err := calculator.Divide(tc.nums...)
+		errReceived := err != nil
+
+		if tc.errExpected != errReceived {
+			fmt.Printf("Expected test: %s\n", tc.name)
+			t.Fatalf("Error: unexpected error status: %v)",
+				errReceived)
+		}
+
+		if !tc.errExpected && tc.want != got {
+			fmt.Printf("Expected test: %s\n", tc.name)
+			t.Errorf("want %g(%T), got %g(%T)", tc.want, tc.want, got, got)
+		}
+	}
+
+}
+
+func TestDivideRandom(t *testing.T) {
+	t.Parallel()
+
+	for i := 0; i < 1000; i++ {
+		var a float64 = rand.Float64()
+		var b float64 = rand.Float64()
+		errExpected := false
+
+		if b == 0 {
+			errExpected = true
+		}
+
+		got, err := calculator.Divide(a, b)
+		errReceived := err != nil
+
+		if errExpected != errReceived {
+			t.Fatalf("Unexpected Error Status: %v", errReceived)
+		}
+
+		want := a / b
+
+		if !errReceived && want != got {
+			t.Errorf("want %.2f, got %.2f", want, got)
+		}
+	}
+}
+
+func TestSqrt(t *testing.T) {
+	t.Parallel()
+
+	testCases := []testCase{
+		{a: 100, want: 10, errExpected: false, name: "Square root of 100 is 10"},
+		{a: 0, want: 0, errExpected: false, name: "Square root of 0 is 0"},
+		{a: -1, want: 999, errExpected: true, name: "Square root can only be calculated for positive numbers"},
+	}
+
+	for _, tc := range testCases {
+		got, err := calculator.Sqrt(tc.a)
+		errReceived := err != nil
+
+		if tc.errExpected != errReceived {
+			t.Fatalf("Unexpected Error - Expected %v, received %v", tc.errExpected, errReceived)
+		}
+
+		if tc.want != got {
```

Comment:
We should only do this comparison if `!errReceived`, shouldn't we?

Reply:
Addressed in 1f30bd8